gulpfile.js

```diff
@@ -0,0 +1,872 @@
+            contents = contents.replace(/\s+$/gm, '').replace(/^\s+/gm, '').replace(/\n+/gm, '\n');
+            contents += '\n';
+        }
+
+        contents = contents.split('"').join('\\"').replace(/\n/gm, '\\n\\\n');
+        contents = copyrightComments + '\
+    //This file is automatically rebuilt by the Cesium build process.\n\
+    /*global define*/\n\
+    define(function() {\n\
+    "use strict";\n\
+    return "' + contents + '";\n\
+});';
+
+        fs.writeFileSync(jsFile, contents);
+    });
+
+    // delete any left over JS files from old shaders
+    Object.keys(leftOverJsFiles).forEach(rimraf.sync);
+
+    var generateBuiltinContents = function(contents, builtins, path) {
+        var amdPath = contents.amdPath;
+        var amdClassName = contents.amdClassName;
+        var builtinLookup = contents.builtinLookup;
+        for (var i = 0; i < builtins.length; i++) {
+            var builtin = builtins[i];
+            amdPath = amdPath + ',\n        \'./' + path + '/' + builtin + '\'';
+            amdClassName = amdClassName + ',\n        ' + 'czm_' + builtin;
+            builtinLookup = builtinLookup + ',\n        ' + 'czm_' + builtin + ' : ' + 'czm_' + builtin;
+        }
+        contents.amdPath = amdPath;
+        contents.amdClassName = amdClassName;
+        contents.builtinLookup = builtinLookup;
+    };
+
+//generate the JS file for Built-in GLSL Functions, Structs, and Constants
+    var contents = {amdPath : '', amdClassName : '', builtinLookup : ''};
+    generateBuiltinContents(contents, builtinConstants, 'Constants');
+    generateBuiltinContents(contents, builtinStructs, 'Structs');
+    generateBuiltinContents(contents, builtinFunctions, 'Functions');
+
+    contents.amdPath = contents.amdPath.replace(',\n', '');
+    contents.amdClassName = contents.amdClassName.replace(',\n', '');
+    contents.builtinLookup = contents.builtinLookup.replace(',\n', '');
+
+    var fileContents = '\
+//This file is automatically rebuilt by the Cesium build process.\n\
+/*global define*/\n\
+define([\n' +
+                       contents.amdPath +
+                       '\n    ], function(\n' +
+                       contents.amdClassName +
+                       ') {\n\
+                           "use strict";\n\
+                           return {\n' + contents.builtinLookup + '};\n\
+});';
+
+    fs.writeFileSync(path.join('Source', 'Shaders', 'Builtin', 'CzmBuiltins.js'), fileContents);
+}
+
+function createCesiumJs() {
+    var moduleIds = [];
+    var parameters = [];
+    var assignments = [];
+
+    var nonIdentifierRegexp = /[^0-9a-zA-Z_$]/g;
+
+    globby.sync(sourceFiles).forEach(function(file) {
+        file = path.relative('Source', file);
+        var moduleId = file;
+        moduleId = filePathToModuleId(moduleId);
+
+        var assignmentName = path.basename(file, path.extname(file));
+        assignmentName = "['" + assignmentName + "']";
+        if (moduleId.indexOf('Source/Shaders/') === 0) {
```

Comment:
This check doesn't work.  `Source/` is not part of the `moduleId`.  Should just be checking for `indexOf('Shaders/')